R/create_query_data.R

```diff
@@ -0,0 +1,894 @@
+#' Creates queries dataset from SMQ and SDG database or custom definitions as
+#' input dataset for `derive_vars_query()`
+#'
+#' Creates queries dataset from definitions in SMQ and SDG database or custom
+#' definitions as input dataset for `derive_vars_query()`.
+#'
+#' @param queries List of queries
+#'
+#'   A list of `query()` objects is expected.
+#'
+#' @param meddra_version MedDRA version
+#'
+#'   The MedDRA version used for coding the terms in the AE dataset should be
+#'   specified. If any of the queries is a SMQ or a customized query including a
+#'   SMQ, the parameter needs to be specified.
+#'
+#'   *Permitted Values*: A character string
+#'
+#' @param whodd_version WHO Drug Dictionary version
+#'
+#'   The version of the WHO Drug Dictionary used for coding the terms in the CM
+#'   dataset should be specified. If any of the queries is a SDG, the parameter
+#'   needs to be specified.
+#'
+#'   *Permitted Values*: A character string
+#'
+#' @param get_smq_fun Function which returns the terms of a SMQ
+#'
+#'   For each query specified for the `queries` parameter which refers to a SMQ
```

Comment:
That would not be correct because the function is also required for customized queries which are based on one or more SMQs, e.g., if a CQ is defined by two SMQs.

I have added some details on when a query refers to an SMQ.